src/Widgets/WifiMenuItem.vala

```diff
@@ -158,20 +158,41 @@ public class Network.WifiMenuItem : Gtk.ListBoxRow {
 
         hide_item (error_img);
         spinner.active = false;
-
-        connect_button_revealer.reveal_child = true;
+        connect_button_revealer.reveal_child = false;
```

Comment:
Connect button should be hidden by default (if we know there is a problem or the device is not usable, we should not let the user infinitely click on something broken).